homeassistant/config_entries.py

```diff
@@ -1117,6 +1119,7 @@ async def async_forward_entry_setup(self, entry: ConfigEntry, domain: str) -> bo
         setup of a component, because it can cause a deadlock.
         """
         # Setup Component if not set up yet
+        domain = platform.value if isinstance(platform, Platform) else platform
```

Comment:
```suggestion
```
Not necessary. The beauty of `StrEnum` is that in basically all relevant respects it behaves just like a normal `str`. E.g. in comparisons, for dictionary keys, in format strings, ...
The only noticeable difference I'm aware of is the `__repr__` method.

Reply:
It is not necessary, but I think it helps make it explicit.
If a method expects a domain we should avoid passing an enum.

Reply:
I agree with @cdce8p, the isinstance check is not needed here.

Reply:
> It is not necessary, but I think it helps make it explicit.
> If a method expects a domain we should avoid passing an enum.

The methods expects a `str` and any subclass will do, `StrEnum` being one of them.
Doing the `isinstance` check will also make typing methods further done the call chain a bit more difficult. Just suppose at some point those should also switch from `str` to `Platform`, we would need to go back and remove all conversions which were unnecessary to begin with.

Reply:
I know that technically the conversion from Platform to string is not necessary, but it was a way to explicitely convert it from a Platform to a Domain. (Explicit is better than implicit some like to point out)